### ISSUE TITLE
Audio delay on DSi LL

### ISSUE THREAD

**Issue by danyisill** [Gericom/GBARunner2#77]
https://www.youtube.com/watch?v=_AZuSBAJvMA
(i manually synchronised the audio so it looks a bit better than it actually is)
As you can see the actual game has nearly no input lag, but the audio is like 1 second behind
the video is a bit old but i just checked out the latest build and the issue still presents

**Comment by Gericom** [comment]
Yeah, this is a know issue. I'm working on something that improves this a lot on musicplayer 2000 (sappy) games. (I think super mario advance 1 also uses that) In general however this is a kinda difficult issue, One processor has to ask the other processor for data because it can't see that memory region. This is not 100% reliable, as in, the other processor can be busy with something that blocks this request for a bit and such. Therefore I need to buffer sound to be able to play it smoothly, but this adds delay. When audio blocks are completely dropped the buffer also starts to desync which causes even more delay.

**Comment by tomudo** [comment]
Gericom , is it  possibly active DSi CPU 133 MHz mode and use those horse power to help audio problem?

**Comment by RocketRobz** [comment]
We're already using 133 Mhz mode, but the issue still persists.

**Comment by Gericom** [comment]
That doesn't help. The issue is just that requesting data from the arm9 via an irq is not reliable enough for low latency.
I basically would need to be able to invoke an arm9 dma transfer from arm7, which is not possible as far as I know.
